File: backend/ProcesosApi/Models/Pago.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace ProcesosApi.Models;

public class Pago
{
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();
    public DateTime Fecha { get; set; }
    public decimal Monto { get; set; }
    public string? Concepto { get; set; }
    public string? MetodoPago { get; set; }
    public DateTime FechaCreacion { get; set; }
}

File: frontend/src/types/pago.ts
export interface Pago {
  id: string;
  procesoId: string;
  fecha: string;
  monto: number;
  concepto: string | null;
  metodoPago: string | null;
  fechaCreacion: string;
}

export interface PagoCreate {
  fecha: string;
  monto: number;
  concepto?: string;
  metodoPago?: string;
}
